istioctl/pkg/kubernetes/client.go

```diff
@@ -195,6 +210,15 @@ func (client *Client) ExtractExecResult(podName, podNamespace, container string,
 
 // GetIstioPods retrieves the pod objects for Istio deployments
 func (client *Client) GetIstioPods(namespace string, params map[string]string) ([]v1.Pod, error) {
+	if client.Revision != "" {
+		labelSelector, ok := params["labelSelector"]
+		if ok {
+			params["labelSelector"] = fmt.Sprintf("%s,istio.io/rev=%s", labelSelector, client.Revision)
+		} else {
+			params["labelSelector"] = fmt.Sprintf("istio.io/rev=%s", client.Revision)
```

Comment:
use the constant, model.RevisionLabel (this should be replaced with the API addition you added, whichever merges first)